kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -410,20 +411,29 @@ def kompilation_up_to_date() -> bool:
 
         return old_digest == kompilation_digest()
 
+    def kompile_haskell() -> None:
+        _LOGGER.info(f'Kompiling definition: {foundry_main_file}')
+        _kompile(foundry_definition_dir, KompileTarget.HASKELL)
+
+    def kompile_llvm() -> None:
+        _LOGGER.info(f'Kompiling definition to LLVM dy.lib: {foundry_main_file}')
+        _kompile(
+            foundry_llvm_dir,
+            KompileTarget.LLVM,
+            llvm_kompile_type=LLVMKompileType.C,
+        )
+
     def update_kompilation_digest() -> None:
         digest_file = foundry_definition_dir / 'digest'
         digest_file.write_text(kompilation_digest())
 
     if not kompilation_up_to_date() or rekompile or not kompiled_timestamp.exists():
-        _LOGGER.info(f'Kompiling definition: {foundry_main_file}')
-        _kompile(foundry_definition_dir, KompileTarget.HASKELL)
-        if llvm_library:
-            _LOGGER.info(f'Kompiling definition to LLVM dy.lib: {foundry_main_file}')
-            _kompile(
-                foundry_llvm_dir,
-                KompileTarget.LLVM,
-                llvm_kompile_type=LLVMKompileType.C,
-            )
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            executor.submit(kompile_haskell)
+            if llvm_library:
+                future2 = executor.submit(kompile_llvm)
+                if future2.exception():
+                    _LOGGER.error(f'Error in kompile_llvm: {future2.exception()}')
```

Comment:
```suggestion
            futures: list[Future] = []
            futures.append(executor.submit(kompile_haskell))
            if llvm_library:
                futures.append(executor.submit(kompile_llvm))
            [future.result() for future in futures]
```

Reply:
Here, `future.result()` takes care of raising the exception on the thread.

Reply:
Done!